OpenMarket/OpenMarket/Controller/AddProductViewController.swift

```diff
@@ -2,19 +2,162 @@
+        self.navigationItem.leftBarButtonItem = cancelItem
+    }
+    
+    private func configureAddImageButton() {
+        self.productManageView.photoAddButton.addTarget(self, action: #selector(imageAddButtonPressed), for: UIControl.Event.touchUpInside)
+    }
+    
+    private func configureImagePicker() {
+        self.imagePicker.sourceType = .photoLibrary
+        self.imagePicker.delegate = self
+    }
+    
+    @objc private func cancelButtonPressed() {
+        self.navigationController?.popViewController(animated: true)
+    }
+    
+    @objc private func doneButtonPressed() {
+        self.uploadNewProduct()
+    }
+    
+    @objc private func imageAddButtonPressed(_ sender: UIButton) {
+        guard imageCount < 5 else { return }
+        self.present(self.imagePicker, animated: true)
     }
+}
+
+extension AddProductViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        if let image = info[UIImagePickerController.InfoKey.originalImage] as? UIImage {
+            self.productManageView.addImageView(with: image)
+        }
+        imageCount += 1
+        dismiss(animated: true, completion: nil)
+    }
+    
+}
+
+extension AddProductViewController: UITextViewDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        UIView.animate(withDuration: 0.4) {
+            self.productManageView.imageScrollView.isHidden = true
+            self.textFieldConstraint =  self.productManageView.textFieldStackView.topAnchor.constraint(equalTo: self.view.safeAreaLayoutGuide.topAnchor)
+            self.textFieldConstraint?.isActive = true
+            self.view.layoutIfNeeded()
+        }
+    }
+    
+    func textViewDidEndEditing(_ textView: UITextView) {
+        self.textFieldConstraint?.isActive = false
+        self.productManageView.imageScrollView.isHidden = false
+        self.view.layoutIfNeeded()
+    }
+    
+    func initializeHideKeyBoard() {
+        let tap: UIGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(dismissMyKeyBoard))
+        self.productManageView.addGestureRecognizer(tap)
+    }
+    
+    @objc func dismissMyKeyBoard() {
+        self.productManageView.endEditing(true)
+    }
+}
 
+extension AddProductViewController {
+    
+    private func fetchNewProductInfo() -> NewProductInfo? {
+        guard let name = self.productManageView.productNameTextField.text, !name.isEmpty,
+              let newProductDescription = self.productManageView.descriptionTextView.text, !newProductDescription.isEmpty,
+              let price = Int(self.productManageView.productPriceTextField.text ?? "0"),
+              let currency = self.productManageView.productPriceSegment.titleForSegment(at: self.productManageView.productPriceSegment.selectedSegmentIndex)
+        else { return nil }
```

Comment:
브레이크포인트를 else절에 찍고 싶으면 어떻게 해야하나요?
```suggestion
        else { 
            return nil 
        }
```
위와 같이 코드를 짜야 명확하게 찍을 수 있지 않을까요?